src/main/scala/firrtl/passes/CheckHighForm.scala

```diff
@@ -81,9 +83,24 @@ trait CheckHighFormLike { this: Pass =>
       m => errors.append(new ModuleNameNotUniqueException(m.info, m.name))
     }
 
+    val extmoduleCollidingPorts = c.modules.collect {
+      case a: ExtModule => a
+    }.groupBy(a => (a.defname, a.params)).mapValues {
```

Comment:
Why look at the params here?

Reply:
Since the parameters might be used to parameterize port widths, then it's only safe to check that extmodules with the same defname and params must have exactly the same ports.

An additional, complementary check would be that all defnames have the same port names, but widths are ignored.

Do you think it makes sense to add that?

Reply:
Makes sense to me; I guess I was assuming the second check (based on the idea that normal Verilog-2005 params can't add/remove ports), but I was mostly wondering if this could get expensive. I guess there aren't normally a ton of blackboxes, so the "hashing types" problem isn't too bad.